src/main/java/games/strategy/triplea/ui/logic/RouteCalculator.java

```diff
@@ -25,57 +35,33 @@ public RouteCalculator(final boolean isInfiniteX, final boolean isInfiniteY, fin
    * Algorithm for finding the shortest path for the given Route.
    *
    * @param route The joints on the Map
-   * @return A Point array which goes through Map Borders if necessary
+   * @return A Point array which goes through Map Borders if necessary or null if route is null
```

Comment:
Doesn't this method now throw an NPE if `route` is `null`?

Reply:
Correct, good catch.
I actually annotated this Method as Nullable before, but then decided this would be stupid because currently no method calls it with a null parameter + (null would be ambigous and trigger a warning)